lizzy/models/stack.py

```diff
@@ -40,81 +28,43 @@ def __init__(self, *,
+        self.status = status
         self.__cf_stack = None
 
-    @staticmethod
-    def generate_version(creation_time: datetime, version: str) -> str:
-        """
-        Generates a version for the stack based on the image version and timestamp.
-        It replaces `.`s with `o`s and removes `_`s to return a valid Cloud Formation Identification
-
-        :param creation_time:  Date and time of stack creation
-        :param version: Docker image version to deploy
-        :return: CF stack version
-        """
-        version = version.lower().replace('-snapshot', 's').replace('.', 'o').replace('_', '')
-        return '{version}T{time:%Y%m%d%H%M%S}'.format(version=version,
-                                                      time=creation_time)
-
-    def generate_id(self) -> str:
-        """
-        The id will be the same as the stack name on aws
-        """
-        return '{name}-{version}'.format(name=self.stack_name, version=self.stack_version)
-
     @classmethod
-    def get(cls, *args, **kwargs) -> "Stack":
-        try:
-            return super().get(*args, **kwargs)
-        except KeyError:
-            raise ObjectNotFound(args[0])
+    def get(cls, stack_name: str, stack_version: str) -> 'Stack':
+        stacks = cls.list(stack_name, stack_version)
+        if not stacks:
+            raise ObjectNotFound('{}-{}'.format(stack_name, stack_version))
+        else:
+            return stacks[0]
 
-    @property
-    def cf_stack(self) -> Dict[str, Any]:
-        """
-        Gets the Cloud Formation Stack
+    @classmethod
+    def list(cls, *stack_ref: List[str]) -> List['Stack']:
         """
+        Returns a List of stack dicts compliant with the API spec.
 
-        if self.__cf_stack is None:
-            cloud_formation = boto3.client('cloudformation',
-                                           region_name=config.region)
-            try:
-                stacks = cloud_formation.describe_stacks(StackName=self.stack_id)  # type: Dict[str, Any]
-            except botocore.exceptions.ClientError:
-                return {}
-            self.__cf_stack = stacks['Stacks'][0]  # type: Dict[str, Any]
+        .. seealso:: lizzy/swagger/lizzy.yaml#/definitions/stack
+        """
 
-        return self.__cf_stack
+        senza = Senza(config.region)
+        stacks = [Stack(**stack)
+                  for stack in senza.list(*stack_ref)]
+        return stacks
 
-    @property
-    def cf_tags(self) -> Dict[str, str]:
+    def generate_id(self) -> str:
         """
-        Return a dict with cloud formation stacks
+        The id will be the same as the stack name on aws
         """
-        kv_tags = self.cf_stack.get('Tags', {})  # type: List[Dict[str, str]]
-        # AWS returns tags in the form of [{'Key': 'KeyName', 'Value': 'KeyValue'}]
-        tags = {tag['Key']: tag['Value'] for tag in kv_tags}
-        return tags
+        return '{name}-{version}'.format(name=self.stack_name,
```

Comment:
This will probably be removed later, In the end I just want to pass the parameters to senza and return the output.